mindmeld/models/embedder_models.py

```diff
@@ -14,119 +14,738 @@
 
     @abstractmethod
-    def load(self, **kwargs):
+    def _load(self, **kwargs):
         """Loads the embedder model
 
         Returns:
             The model object.
         """
-        pass
+
+        # backwards compatability
+        if getattr(self, "load", None):
+            # implies `load()` is implemented as an abstract class instead of the new way `_load()`
+            getattr(self, "load")(**kwargs)
+
+        raise NotImplementedError
 
     @abstractmethod
-    def encode(self, text_list):
+    def _encode(self, text_list):
         """
         Args:
             text_list (list): A list of text strings for which to generate the embeddings.
 
         Returns:
             (list): A list of numpy arrays of the embeddings.
         """
-        pass
 
-    def clear_cache(self):
-        """Deletes the cache file."""
-        if os.path.exists(self.cache_path):
-            os.remove(self.cache_path)
+        # backwards compatability
+        if getattr(self, "encode", None):
+            # implies `encode()` is implemented as an abstract class instead of new way `_encode()`
+            getattr(self, "encode")(text_list)
+
+        raise NotImplementedError
 
-    def get_encodings(self, text_list):
+    # deprecated method
+    def dump(self):
+        msg = f"DeprecationWarning: Use {self.__class__.__name__}.dump_cache() instead of " \
+              f"{self.__class__.__name__}.dump()"
+        logger.warning(msg)
+        self.dump_cache()
+
+    def get_encodings(self, text_list, add_to_cache=True):
         """Fetches the encoded values from the cache, or generates them.
         Args:
             text_list (list): A list of text strings for which to get the embeddings.
+            add_to_cache (bool): If True, adds te encodings to self.cache and returns embeddings
 
         Returns:
             (list): A list of numpy arrays with the embeddings.
         """
         encoded = [self.cache.get(text, None) for text in text_list]
         cache_miss_indices = [i for i, vec in enumerate(encoded) if vec is None]
         text_to_encode = [text_list[i] for i in cache_miss_indices]
-        model_encoded_text = self.encode(text_to_encode)
+        model_encoded_text = self._encode(text_to_encode)
 
         for i, v in enumerate(cache_miss_indices):
             encoded[v] = model_encoded_text[i]
-            self.cache[text_to_encode[i]] = model_encoded_text[i]
+            if add_to_cache:
+                self.cache[text_to_encode[i]] = model_encoded_text[i]
+
         return encoded
 
-    def dump(self):
-        """Dumps the cache to disk."""
-        with open(self.cache_path, "wb") as fp:
-            pickle.dump(self.cache, fp)
+    def add_to_cache(self, superficial_data):
+        """Manually add some embeddings to cache. This method is created to entertain storing
+        superficial text-encoding pairs wherein the encodings are not the encodings of the text
```

Comment:
Can't we be more specific and call it average/max pooled whitelist embeddings? Don't think we plan to modify or provide alternate methods to do this operation anyway.

Reply:
Good point. I'll do that change.